crypto/signer.go

```diff
@@ -31,39 +31,39 @@ var sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
-	hash   crypto.Hash
```

Comment:
Looks like you're exposing previously private member objects. I didn't quite catch the reason?